patches/server/0839-Bucketable-API.patch

```diff
@@ -0,0 +1,69 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Owen1212055 <23108066+Owen1212055@users.noreply.github.com>
+Date: Sun, 26 Dec 2021 14:03:17 -0500
+Subject: [PATCH] Bucketable API
+
+
+diff --git a/src/main/java/io/papermc/paper/entity/PaperBucketable.java b/src/main/java/io/papermc/paper/entity/PaperBucketable.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..9eb22880bf704c74b2cb1379dee5e9b77c345493
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/entity/PaperBucketable.java
+@@ -0,0 +1,31 @@
++package io.papermc.paper.entity;
++
++import org.bukkit.Sound;
++import org.bukkit.craftbukkit.CraftSound;
++import org.bukkit.craftbukkit.inventory.CraftItemStack;
++import org.bukkit.inventory.ItemStack;
++
++public interface PaperBucketable extends Bucketable {
++
++    net.minecraft.world.entity.animal.Bucketable getHandle();
++
++    @Override
++    default boolean isFromBucket() {
++        return this.getHandle().fromBucket();
++    }
++
++    @Override
++    default void setFromBucket(boolean fromBucket) {
++        this.getHandle().setFromBucket(fromBucket);
++    }
++
++    @Override
++    default ItemStack getBaseBucketItem() {
++        return CraftItemStack.asCraftMirror(this.getHandle().getBucketItemStack());
```

Comment:
This should be asBukkitCopy so it doesn't give any wrong ideas about modifying the returned itemstack having an effect on the entities ability to reflect changes (which it doesn't have)